scylla/src/network/connection_pool.rs

```diff
@@ -816,6 +821,23 @@ impl PoolRefiller {
                     }
                 }
 
+                // At this point, the connection is ready to be put into the pool.
+                // This also means that we have at least one working connection.
+                if let Some(ref connectivity_notifier) = self.pool_config.connectivity_events_sender
+                    && self.is_empty()
+                {
+                    // If we didn't have any connectivity with the host previously, we can now
+                    // notify that the host is alive again.
+                    debug!(
+                        "[{}] Connection pool is no longer empty, notifying listeners",
+                        endpoint,
+                    );
+
+                    // Ignore failure to send. If there are no listeners, it's fine.
+                    let event = ConnectivityChangeEvent::Established { address: endpoint };
+                    let _ = connectivity_notifier.send(event);
+                }
```

Comment:
I wonder, why are you performing the notification here?
I see that below there are some checks which could eliminate the connection from being added.
In the case it is accepted, `update_shared_conns` will be called, which also has notification logic added by you. Why duplicate it?

Reply:
`update_shared_conns` only has logic for DOWN, not for UP.